### PR TITLE
Jso contours

### PR DESCRIPTION
**Pull request by jstonge**
Change logs:
- Got rid of all the charting code. This is now in `py-allotax`, as it was hardcoded for that anyway
- `combine_distributions` only export mixedElems now. More generally, the pkg is more streamlined. 

To do:
- [ ] I still need to clean `DiamondChart`. Maybe there's a way to reduce the number of arguments.
- [ ] Right now, we are using a custom `alpha_norm_type2` in the `make_grid` function. This need to disappear. Ideally, we would only call that function from the main file that has been tested.
  - `alpha_norm_type2` is required in there because we need it to evaluate the grid we use to draw contourlines. We cannot use the main function right now because the main function is doing the following: `rank_turbulence_divergence(mixedelements, alpha) `. That is, it expects `mixedelements`. We might want to rethink that logic.

p.s. This update goes hand in hand with [this commit](https://github.com/carterwward/py-allotax/blob/jso-experimental/src/AllotaxChart.js) in py-allotax. When we merge this commit, we need to update `py-allotax` as well. I don't like that.

### CODE REVIEW

**Comment by amafehr** [comment]
Suggest removing `node_modules/`

File: package-lock.json
Comment (amafehr) [inline package-lock.json:1]: Is this file needed?

File: .gitignore
Comment (amafehr) [inline .gitignore:1]: add `*node_modules/`

Comment (amafehr) [reply .gitignore:1]: likewise `*data/` ignores everything inside `data/`, so you can remove `data/data/...`

File: src/diamond_count.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ function diamond_counts(mixedelements) {
 
 
 // we expect wordshift to be of the form { divergence_elements: [ length of type ], normalization: float }
-export default function diamond(mixedelements, wordshift) {
+export default function diamond_count(mixedelements, wordshift) {
   
   // let rank_turbulence = rank_turbulence_divergence(mixedelements, alpha)
```
Comment (amafehr) [inline src/diamond_count.js:88]: remove commented lines. If there's a reason to keep them, write out a comment.

File: src/utils_helpers.js [hunk 2]
```diff
@@ -82,3 +101,37 @@ function zeros(length){
...
+    const rank_diff = me[0]['ranks'][i]-me[1]['ranks'][i]
+    out.push({
+      'type': `${me[0]['types'][i]} (${me[0]['ranks'][i]} ⇋ ${me[1]['ranks'][i]})` ,
+      'rank_diff': rank_diff,
+      'metric': rank_diff < 0 ? -dat.deltas[i] : dat.deltas[i], 
+    })
+  }
+  
+  return out.slice().sort((a, b) => descending(Math.abs(a.metric), Math.abs(b.metric)))
+}
```
Comment (amafehr) [inline src/utils_helpers.js:135]: These helpers don't feel general enough to go in `utils`. I suggest putting them in `combine_distributions.js` because they're operating specifically on mixed elements calculations.

File: src/alpha_norm_type2.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export default function alpha_norm_type2(x1, x2, alpha) {
```
Comment (amafehr) [inline src/alpha_norm_type2.js:1]: It's a bit odd by itself so I suggest we add context on what it's for:
```suggestion
// adapted from matlab, normalizes ____
export default function alpha_norm_type2(x1, x2, alpha) {
```

File: README.md [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
-<img width="100%" alt="pipeline" src="https://github.com/user-attachments/assets/11ebfb8d-95d7-43f1-b95d-2f14bf4dac86">
+<img width="100%" alt="pipeline" src="https://github.com/user-attachments/assets/eee02d08-9c76-4a6d-a103-812d7cac81a1">
 <p>
 
-- https://observablehq.com/@jstonge/allotaxonometer-for-all-exp
-- https://allotaxp.vercel.app/
-- :octocat: https://github.com/jstonge/allotaxp
-- :octocat: https://github.com/carterwward/py-allotax
+- Visit our [single-page web application](https://allotax.vercel.app/) to try it out online ([:octocat: github](https://github.com/jstonge/allotax))!
+- Use [py-allotax](https://github.com/carterwward/py-allotax) to use allotaxonometry at scale ([:octocat: github](https://github.com/carterwward/py-allotax))
```
Comment (amafehr) [inline README.md:6]: Move the tools under the description of "Allotaxonometer.js is a lightweight ..."

File: README.md [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
 
-The primary purpose of the alloxonometer in general is to calculate and visualize the difference between any two Zipfian ranked lists of components. This package provides the utilities to facilitate the creationg of the allotaxonometer in `d3.js`
+`Alloxonometer.js` is a lightweigt Javascript package for combining pairs of Zipfian ranked lists of components and provide [divergence metrics](https://compstorylab.org/allotaxonometry/papers/rank-turbulence-divergence/).
```
Comment (amafehr) [inline README.md:8]:
```suggestion
`Alloxonometer.js` is a lightweigt Javascript package for combining pairs of Zipfian ranked lists of components and provide [divergence metrics](https://compstorylab.org/allotaxonometry/papers/rank-turbulence-divergence/). The library performs the divergence calculations among two systems, which will typically be consumed by the following 2 tools to visualize results.
```

File: README.md [hunk 6]
```diff
@@ -1,13 +1,11 @@
...
```
Comment (amafehr) [inline README.md:9]: I suggest we add in some of the long foretold documentation (unless you want to do that separately) and close out issues related to updating the docs: "add documentation for users (disambiguate among repos and what they're doing), highlighting tools per use case and extensibility. "

Edit below, just generally:
```suggestion
## Use cases and install
- Explore systems and formulate research questions: we recommend starting with the web app...
- Running multiple or larger scale analyses: we recommend using the py-allotax, which details its install instructions, details the required data format, and provides examples in its repo.
```

**Review by amafehr — COMMENTED** [review]
Generally makes sense to me given the discussion of moving plotting code to their respective tools (given how customized they're become). 

A few notes:
- I think we should throw in more documentation while we're at it. See my comments. I am not sure if there's a use case for researchers directly using calculations from `allotaxonometer.js`--if you think there is, we need to add install instructions specific to this repo.
- I suggest moving a few functions around so we don't lose the thread on what little functions are meant to do.

File: src/utils_helpers.js (hunk 2 above)
Comment (jstonge) [reply src/utils_helpers.js]: done

File: src/alpha_norm_type2.js (hunk 3 above)
Comment (jstonge) [reply src/alpha_norm_type2.js:12]: done
